search/FindPeakElement.cpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * \file FindPeakElement.cpp
+ * \brief [Find peak element in Array]
+ * @author Unknown author
+ * @author https://github.com/AbhineshJha 
+*/
+
+#include <cassert>   // for assert
+#include <iostream>  // for IO operations
+
+/**
+ * \brief [Algorithm implementation for finding a peak element]
+ * \param [in] array array to search in
+ * \param [in] size length of array
+ * \returns index of a peak element in the array
+ * \returns -1 if no peak element is found
+ */
+int FindPeakElement(int *array, int size) {
+    if (size == 0) {
+        return -1; // No elements in the array
+    }
+
+    if (size == 1) {
+        return 0;  // Single element is a peak
+    }
+
+    // Check the first element
+    if (array[0] >= array[1]) {
+        return 0;
+    }
+
+    // Check the last element
+    if (array[size - 1] >= array[size - 2]) {
+        return size - 1;
+    }
+
+    // Check elements in the middle
+    for (int i = 1; i < size - 1; ++i) {
```

Comment:
Hi @AbhineshJha  , if there is only one peak element then we can find this element in O(logN) time using binary search

Reply:
done